src/icu/break_iterator.cr

```diff
@@ -24,20 +24,17 @@ class ICU::BreakIterator
     @ubrk = LibICU.ubrk_open(break_type, locale, nil, 0, pointerof(ustatus))
     ICU.check_error!(ustatus) { free }
 
-    ustatus = LibICU::UErrorCode::UZeroError
-    @utext = LibICU.utext_open_ut_f8(nil, @text.to_unsafe, @text.bytesize, pointerof(ustatus))
-    ICU.check_error!(ustatus) { free }
+    @chars = text.chars.map { |c| c.ord.to_u16 }
```

Comment:
Can you replace `text.chars.map` by `text.each_char.map` here (this lazy approach uses an Iterator so no intermediate array of chars is created)